engines/private/private.cpp

```diff
@@ -0,0 +1,1120 @@
+    Common::File *file = new Common::File();
+    Common::String path = convertPath(name);
+
+    if (!file->open(path))
+        error("unable to find video file %s", path.c_str());
+
+    if (!_videoDecoder->loadStream(file))
+        error("unable to load video %s", path.c_str());
+    _videoDecoder->start();
+
+}
+
+void PrivateEngine::skipVideo() {
+    _videoDecoder->close();
+    delete _videoDecoder;
+    _videoDecoder = nullptr;
+}
+
+void PrivateEngine::stopSound(bool all) {
+    debugC(3, kPrivateDebugExample, "%s", __FUNCTION__);
+
+    if (all) {
+        _mixer->stopHandle(_fgSoundHandle);
+        _mixer->stopHandle(_bgSoundHandle);
+    }
+    else {
+        _mixer->stopHandle(_fgSoundHandle);
+    }
+}
+
+void PrivateEngine::loadImage(const Common::String &name, int x, int y) {
+    debugC(1, kPrivateDebugExample, "%s : %s", __FUNCTION__, name.c_str());
+    Common::File file;
+    Common::String path = convertPath(name);
+    if (!file.open(path))
+        error("unable to load image %s", path.c_str());
+
+    _image->loadStream(file);
+    Graphics::Surface *surf = _image->getSurface()->convertTo(_pixelFormat, _image->getPalette());
+    _compositeSurface->transBlitFrom(*surf, *_origin + Common::Point(x,y), _transparentColor);
+    surf->free();
+    delete surf;
+    _image->destroy();
+    drawScreen();
+}
+
+void PrivateEngine::drawScreenFrame(Graphics::Surface *screen) {
+    Common::String path = convertPath(*_frame);
+    Common::File file;
+    assert(file.open(path));
+    _image->loadStream(file);
+    Graphics::Surface *csurf = _image->getSurface()->convertTo(_pixelFormat, _image->getPalette());
+    screen->copyRectToSurface(*csurf, 0, 0, Common::Rect(0, 0, _screenW, _screenH));
+    csurf->free();
+    delete csurf;
+    _image->destroy();
+}
+
+
+Graphics::ManagedSurface *PrivateEngine::loadMask(const Common::String &name, int x, int y, bool drawn) {
+    debugC(1, kPrivateDebugExample, "%s : %s", __FUNCTION__, name.c_str());
+    Common::File file;
+    Common::String path = convertPath(name);
+    if (!file.open(path))
+        error("unable to load mask %s", path.c_str());
+
+    _image->loadStream(file);
+    Graphics::ManagedSurface *surf = new Graphics::ManagedSurface();
+    Graphics::Surface *screen = g_system->lockScreen();
+
+    surf->create(_screenW, _screenH, _pixelFormat);
+    surf->transBlitFrom(*screen);
+    g_system->unlockScreen();
+    Graphics::Surface *csurf = _image->getSurface()->convertTo(_pixelFormat, _image->getPalette());
+    surf->transBlitFrom(*csurf, Common::Point(x,y));
+    csurf->free();
+    delete csurf;
+    _image->destroy();
+
+    if (drawn) {
+        drawMask(surf);
+    }
+
+    return surf;
+}
+
+void PrivateEngine::drawMask(Graphics::ManagedSurface *surf) {
+    _compositeSurface->transBlitFrom(surf->rawSurface(), *_origin, _transparentColor);
+    drawScreen();
+}
+
+void PrivateEngine::drawScreen() {
+    Graphics::Surface *screen = g_system->lockScreen();
```

Comment:
You need to lock screen like this only if you have direct access to it and you're calling only `copyRectToScreen`. Thus, both lock/unlock calls could be safely removed.

Reply:
Fixed. If you have more advice on how to improve performance, please let me know. I still notice some flickering.